examples/with-cloudinary/pages/index.tsx

```diff
@@ -0,0 +1,176 @@
+import type { NextPage } from 'next'
+import Head from 'next/head'
+import Image from 'next/image'
+import Link from 'next/link'
+import { useRouter } from 'next/router'
+import { useEffect } from 'react'
+import Bridge from '../components/Icons/Bridge'
+import Logo from '../components/Icons/Logo'
+import Modal from '../components/Modal'
+import cloudinary from '../utils/cloudinary'
+import getBase64ImageUrl from '../utils/generateBlurPlaceholder'
+import type { ImageProps } from '../utils/types'
+import { useLastViewedPhoto } from '../utils/useLastViewedPhoto'
+
+const Home: NextPage = ({ images }: { images: ImageProps[] }) => {
+  const router = useRouter()
+  const { photoId } = router.query
+  const [lastViewedPhoto, setLastViewedPhoto] = useLastViewedPhoto()
+
+  useEffect(() => {
+    // This effect keeps track of the last viewed photo in tshe modal to keep the index page in sync when the user navigates back
+    if (lastViewedPhoto && !photoId) {
+      document
+        .querySelector(`#photo-${lastViewedPhoto}`)
```

Comment:
Is a `querySelector` the way we want to do this? Or could / should it be done using a `ref`?

Reply:
I actually paired with Sam/Ryan on this and suggested using a `ref` but they recommended I do it with `querySelector` instead. Can't remember the reason why – perhaps speed? Cause otherwise, I'd have to attach a ref to each of the 300+ images on a page.

Reply:
You think I should explore using `ref` instead?

Reply:
Yeah, could be a lot of refs floating around. 🤔 Just don't see query selecting often. Made me do a double take and ask. Perhaps @ijjk and @styfle have better ideas.

Reply:
I agree – I'll add this to my list of improvements for early next week. Sound good?

So far I have:
- Add infinite scrolling
- Fix bottom photo navbar animations on very wide screens
- Convert querySelector to `ref`

For now, I'd love to ship this since we're planning to send the email tomorrow morning.

Reply:
`querySelector` seems really odd with react

Lets compare the perf with `ref